### ISSUE TITLE
Washing doesn't recalculate requirements on lowering item count

### ISSUE THREAD

**Issue by Brambor** [CleverRaven/Cataclysm-DDA#50431]
### Describe the bug

Requirements for washing are not recalculated when lowering item count specific way.

### Steps To Reproduce

1. Make a player with a washing board and a bunch of filthy rags.
2. [a]ctivate the washing board and select all rags for cleaning.
3. Without deselecting rags, select one filthy rag (press 1, then right arrow on rags).
4. Observe the required water and the cleanser stayed the same.

### Expected behavior

Recalculation on lowering item count. It is currently recalculated only on increasing item count.

### Screenshots

![CDDA washing recalculation](https://user-images.githubusercontent.com/13402666/127913443-33100458-c744-45a6-a8c7-fea8df14f4d2.png)

### Versions and configuration

- OS: Windows 10
- Game Version: 2021-08-02-0846
- Graphics version: Tiles

**Comment by hexagonrecursion** [comment]
If you hit enter after step 4 the game washes all the rags even though it says that it will only wash one.

**Comment by ZeroInternalReflection** [comment]
I'm still familiarizing myself with the relevant sections of code, so I don't know if this is the same bug or a parallel one, but the 'd'rop menu has similar behaviour (tested on cdda-experimental-2021-08-16-1305).
**Test 1:**
1. Select a stack of items to drop
2. Tell it to only drop 1 instead
3. UI says it will drop 1, but doesn't update volume/weight change
4. Drops the entire stack

**Test 2:**
1. Select 1 item from a stack to drop
2. Tell it to drop 2 items instead-
3. I says it will drop 2, and updates volume/weight change
4. Drops 2

The 'g'et menu, on the other hand, operates differently:
1. Prepare to get 1 item in a stack
2. Tell it to get 2 instead
3. It ignores you.  No UI change.  Still getting 1 item.  The UI and the results of the action only change when you deselect the entire stack.

It seems like the dropping/washing/(and perhaps other similar actions?) code, when you tell it to select x items, only makes sure that _at least_ x items are selected, and doesn't make sure _only_ x items are selected, only properly clearing that out when you deselect the entire stack.

**Comment by Brambor** [comment]
@ZeroInternalReflection Yust a nag: If you were able to merge these similar code bits into one function, it would be amazing!

**Comment by ZeroInternalReflection** [comment]
So, it looks like the washing and dropping are two separate functions (inventory_iuse_selector::set_chosen_count and inventory_drop_selector::set_chosen_count) that happened to have the same quirk.  They would loop through the stack until they reached 'count', flagging items for washing/dropping, then break.  So, if 'count' had previously been higher, the later items in the stack remained flagged.

I've made a couple of test changes that seem to resolve both issues (see above).  With these changes, the items in a stack should be unflagged every time the number selected changes.  It would probably be cleaner to continue the loop past the current break, unflagging everything else, but for reasons I'm not entirely clear on, that didn't work when I tried it.

Trying to unify the two functions (and ensuring the 'g'et menu has matching behaviour) is going to require a bit more thought and testing.
